source/MaterialXView/ShaderValidators/Osl/OslValidator.h

```diff
@@ -89,6 +89,15 @@ class OslValidator : public ShaderValidator
         _oslIncludePathString = includePathString;
     }
 
+    /// Set OSL output name, excluding any extension.
+    /// During compiler checking an OSL file of the given output name will be used if it
+    /// is not empty. If temp then OSL will be written to a temporary file.
+    /// @param filePathString Full path name
+    void setOslOutputFilePath(const std::string filePathString)
```

Comment:
Allow setting a output file name path for the osl file. We need this since all we currently pass over is the ShaderPtr which has no info which can be used to extract out the osl file name we want.